### PR TITLE
add Aperture from Swiss AI

### PR DESCRIPTION
**Pull request by Goekdeniz-Guelmez**

### CODE REVIEW

**Comment by Goekdeniz-Guelmez** [comment]
(mlx-dev)  ~/Desktop/mlx-lm/ [add-swiss-ai-Apertus*] python -m mlx_lm.generate --model swiss-ai/Apertus-8B-Instruct-2509 --prompt "Who is Michael Jackson"
<frozen runpy>:128: RuntimeWarning: 'mlx_lm.generate' found in sys.modules after import of package 'mlx_lm', but prior to execution of 'mlx_lm.generate'; this may result in unpredictable behaviour
Calling `python -m mlx_lm.generate...` directly is deprecated. Use `mlx_lm.generate...` or `python -m mlx_lm generate ...` instead.
Fetching 12 files: 100%|█████████████████████████████████████| 12/12 [00:00<00:00, 142987.64it/s]
==========
Michael Jackson was an American singer, songwriter, and dancer. He is often referred to as the "King of Pop" and is considered one of the most influential figures in the history of popular music. Jackson's career spanned over four decades, and he is known for his distinctive voice, dance moves, and music videos that pushed the boundaries of entertainment. Some of his most famous songs include "Thriller," "Billie Jean," and "Beat It." Jackson's personal life was also the subject
==========
Prompt: 65 tokens, 104.829 tokens-per-sec
Generation: 100 tokens, 6.429 tokens-per-sec
Peak memory: 16.216 GB

**Comment by Goekdeniz-Guelmez** [comment]
@awni shoud this XieLU activation function get into mlx itself? I also think relu2 shoudl be added since more and more model start to use it.

**Comment by Goekdeniz-Guelmez** [comment]
python -m mlx_lm.lora  --train --model swiss-ai/Apertus-8B-Instruct-2509 --data mlx-community/wikisql --num-layers 4 --iters 100 --batch-size 1 --steps-per-report 20 --max-seq-length 256 --adapter-path /Users/gokdenizgulmez/Desktop/test
Calling `python -m mlx_lm.lora...` directly is deprecated. Use `mlx_lm.lora...` or `python -m mlx_lm lora ...` instead.
Loading pretrained model
Fetching 12 files: 100%|██████████████████████████████████████| 12/12 [00:00<00:00, 40427.03it/s]
Loading datasets
Loading Hugging Face dataset mlx-community/wikisql.
Training
Trainable parameters: 0.005% (0.426M/8053.338M)
Starting training..., iters: 100
Calculating loss...: 100%|███████████████████████████████████████| 25/25 [00:15<00:00,  1.63it/s]
Iter 1: Val loss 2.688, Val took 15.355s
Iter 20: Train loss 2.617, Learning Rate 1.000e-05, It/sec 1.316, Tokens/sec 117.396, Trained Tokens 1784, Peak mem 16.653 GB
Iter 40: Train loss 2.177, Learning Rate 1.000e-05, It/sec 1.387, Tokens/sec 140.110, Trained Tokens 3804, Peak mem 16.656 GB
Iter 60: Train loss 1.811, Learning Rate 1.000e-05, It/sec 1.493, Tokens/sec 135.937, Trained Tokens 5625, Peak mem 16.656 GB
Iter 80: Train loss 1.704, Learning Rate 1.000e-05, It/sec 1.447, Tokens/sec 142.277, Trained Tokens 7591, Peak mem 16.656 GB
Calculating loss...: 100%|███████████████████████████████████████| 25/25 [00:15<00:00,  1.66it/s]
Iter 100: Val loss 1.577, Val took 15.044s
Iter 100: Train loss 1.715, Learning Rate 1.000e-05, It/sec 1.426, Tokens/sec 137.931, Trained Tokens 9525, Peak mem 16.656 GB

**Comment by Goekdeniz-Guelmez** [comment]
@awni should be mergeable now!

**Comment by martinjaggi** [comment]
thanks a lot for pushing this!

**Comment by Goekdeniz-Guelmez** [comment]
of course, this is a great model!

**Review by awni — APPROVED** [review]
Thanks!!
